src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,107 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+const initialMovieState = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   const [count] = useState(0);
+  const [movie, setMovie] = useState(initialMovieState);
+
+  const handlerInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+
+    setMovie(prevInputs => ({ ...prevInputs, [name]: value }));
+  };
+
+  const isValidForm =
+    !movie.title.trim() ||
+    !movie.imgUrl.trim() ||
+    !movie.imdbUrl.trim() ||
+    !movie.imdbId.trim();
+
+  const reset = () => {
+    movie.title = '';
+    movie.description = '';
+    movie.imgUrl = '';
+    movie.imdbUrl = '';
+    movie.imdbId = '';
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (isValidForm) {
+      return;
+    }
+
+    onAdd({
+      ...movie,
+    });
```

Comment:
no needs to make a copy
```suggestion
    onAdd(movie);
```